xbmc/video/VideoDatabase.cpp

```diff
@@ -4864,10 +4864,23 @@ void CVideoDatabase::SetVideoSettings(int idFile, const CVideoSettings &setting)
   }
 }
 
-void CVideoDatabase::SetArtForItem(int mediaId, const MediaType &mediaType, const std::map<std::string, std::string> &art)
+void CVideoDatabase::SetArtForItem(int mediaId,
+                                   const MediaType& mediaType,
+                                   const std::map<std::string, std::string>& art)
 {
-  for (const auto &i : art)
-    SetArtForItem(mediaId, mediaType, i.first, i.second);
+  if (mediaType == MediaTypeMovie)
+  {
+    // Take the hit of looking up the default version of the movie only once, instead of once per
+    // art piece.
+    CFileItem item{};
+    GetDefaultVideoVersion(VideoDbContentType::MOVIES, mediaId, item);
+    SetArtForItem(item.GetVideoInfoTag()->m_iDbId, MediaTypeVideoVersion, art);
+  }
+  else
+  {
+    for (const auto& i : art)
+      SetArtForItem(mediaId, mediaType, i.first, i.second);
```

Comment:
You can simplify this by calling `void SetArtForItem(int mediaId, const MediaType &mediaType, const std::map<std::string, std::string> &art);` - no need to iterate yourself.

Reply:
That would cause infinite recursion. This block is the original code. I added the block above to translate the media_id / media_type parameters and call the original code through 1 level of recursion, it will never go beyond that.
Even that recursion could be avoided, but I didn't find a way that would make extra copies along the way.

Reply:
Alright.